internal/controller/clusteradmissionpolicy_controller.go

```diff
@@ -80,110 +79,35 @@ func (r *ClusterAdmissionPolicyReconciler) SetupWithManager(mgr ctrl.Manager) er
 		For(&policiesv1.ClusterAdmissionPolicy{}).
 		Watches(
 			&corev1.Pod{},
-			handler.EnqueueRequestsFromMapFunc(r.findClusterAdmissionPoliciesForPod),
+			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, object client.Object) []reconcile.Request {
+				return findClusterAdmissionPoliciesForPod(ctx, r.Client, object)
+			}),
 		).
 		// Despite this policy server watch is not strictly necessary, we
 		// include it for the integration tests, so that we identify
 		// policy server creations even when the controller-manager is not
 		// present (so no pods end up being created)
 		Watches(
 			&policiesv1.PolicyServer{},
-			handler.EnqueueRequestsFromMapFunc(r.findClusterAdmissionPoliciesForPolicyServer),
+			handler.EnqueueRequestsFromMapFunc(func(ctx context.Context, object client.Object) []reconcile.Request {
+				return findClusterAdmissionPoliciesForPolicyServer(ctx, r.Client, object, r.DeploymentsNamespace)
+			}),
 		).
 		Watches(
 			&admissionregistrationv1.ValidatingWebhookConfiguration{},
-			handler.EnqueueRequestsFromMapFunc(r.findClusterAdmissionPolicyForWebhookConfiguration),
+			handler.EnqueueRequestsFromMapFunc(func(_ context.Context, webhookConfiguration client.Object) []reconcile.Request {
+				return findClusterAdmissionPolicyForWebhookConfiguration(webhookConfiguration, r.Log)
+			}),
 		).
 		Watches(
 			&admissionregistrationv1.MutatingWebhookConfiguration{},
-			handler.EnqueueRequestsFromMapFunc(r.findClusterAdmissionPolicyForWebhookConfiguration),
+			handler.EnqueueRequestsFromMapFunc(func(_ context.Context, webhookConfiguration client.Object) []reconcile.Request {
+				return findClusterAdmissionPolicyForWebhookConfiguration(webhookConfiguration, r.Log)
+			}),
```

Comment:
I feel sorry because I asked for this change, but I didn't expect the code to become like that. I didn't notice the functions being called need to access some instance attributes of the `r` object.

I think the old code, like the one on the left looked better. Sorry again! :pray: